### PR TITLE
Update seed file and Request memory for secured_message on fuzzing test case

### PR DESCRIPTION
**Pull request by yaohuixguo**
Fix : #870
Fix : #872
Fixed "transport_message_size" problem, similar to "test_spdm_transport_pci_doe_encode_message"
Signed-off-by: yaohuixguo <yaohuix.guo@intel.com>

### CODE REVIEW

File: unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c [hunk 1]
```diff
@@ -56,14 +56,15 @@ void libspdm_test_encode_secured_message(void **State)
                              sizeof(spdm_secured_message_cipher_header_t) +
                              32; /* MCTP_MAX_RANDOM_NUMBER_COUNT */
     LIBSPDM_ASSERT(spdm_test_context->test_buffer_size > record_header_max_size);
-
-    secured_message_size = spdm_test_context->test_buffer_size - record_header_max_size;
+    secured_message_size = spdm_test_context->test_buffer_size;
+    secured_message = (void *)malloc(secured_message_size);
```
Comment (jyao1) [inline unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c:61]: why you need malloc?

File: unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c [hunk 2]
```diff
@@ -93,11 +94,6 @@ void libspdm_run_test_harness(void *test_buffer, size_t test_buffer_size)
         /* buffer too small */
         return;
     }
-    if (buffer_size >
-        LIBSPDM_MAX_MESSAGE_BUFFER_SIZE - record_header_max_size - aead_tag_max_size) {
-        buffer_size = LIBSPDM_MAX_MESSAGE_BUFFER_SIZE - record_header_max_size -
-                      aead_tag_max_size;
-    }
```
Comment (jyao1) [inline unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c:97]: why remove this check?

File: unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c (hunk 1 above)
Comment (yaohuixguo) [reply unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c:61]: 1.secured_ message requires memory to complete encode:
line 266 do assignment operation
https://github.com/DMTF/libspdm/blob/bd5cde40aad7df7d07a54df5b11664d0fb53b1ed/library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c#L261-L266
2.secured_ message is the AUTH session, including the header
https://github.com/DMTF/libspdm/blob/bd5cde40aad7df7d07a54df5b11664d0fb53b1ed/library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c#L252-L255

File: unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c (hunk 2 above)
Comment (yaohuixguo) [reply unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c:97]: test_buffer is the AUTH session, this check puts test_buffer as AppData
https://github.com/DMTF/libspdm/blob/bd5cde40aad7df7d07a54df5b11664d0fb53b1ed/include/industry_standard/spdm_secured_message.h#L37-L52

Comment (jyao1) [reply unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c:97]: If APP data is not consumed, why you care?
Basically, you just fuzz the App_Length.

File: unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c (hunk 1 above)
Comment (jyao1) [reply unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c:61]: 1) libspdm already defined maximum size. Why not use that in stack? Why you have to use heap?

2) If APP data is not consumed, why you care? Basically, you just fuzz the App_Length.

File: unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c [hunk 3]
```diff
@@ -57,12 +57,12 @@ void libspdm_test_encode_secured_message(void **State)
                              32; /* MCTP_MAX_RANDOM_NUMBER_COUNT */
     LIBSPDM_ASSERT(spdm_test_context->test_buffer_size > record_header_max_size);
 
-    secured_message_size = spdm_test_context->test_buffer_size - record_header_max_size;
+    secured_message_size = sizeof(secured_message);
```
Comment (jyao1) [inline unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c:60]: Please explain why.

File: unit_test/fuzzing/test_transport/test_spdm_transport_mctp_encode_message/transport_mctp_encode_message.c [hunk 4]
```diff
@@ -39,7 +43,10 @@ void libspdm_test_transport_mctp_encode_message(void **State)
                              32; /* MCTP_MAX_RANDOM_NUMBER_COUNT */
     LIBSPDM_ASSERT(spdm_test_context->test_buffer_size > record_header_max_size);
 
-    transport_message_size = spdm_test_context->test_buffer_size - record_header_max_size;
+    alignment = MCTP_ALIGNMENT;
+    aligned_message_size = (spdm_test_context->test_buffer_size + (alignment - 1)) &
+                           ~(alignment - 1);
+    transport_message_size = aligned_message_size;
```
Comment (jyao1) [inline unit_test/fuzzing/test_transport/test_spdm_transport_mctp_encode_message/transport_mctp_encode_message.c:49]: Please explain why.

File: unit_test/fuzzing/test_transport/test_spdm_transport_pci_doe_encode_message/spdm_transport_pci_doe_encode_message.c [hunk 5]
```diff
@@ -39,7 +43,10 @@ void libspdm_test_transport_pci_doe_encode_message(void **State)
                              0; /* PCI_DOE_MAX_RANDOM_NUMBER_COUNT */
     LIBSPDM_ASSERT(spdm_test_context->test_buffer_size > record_header_max_size);
 
-    transport_message_size = spdm_test_context->test_buffer_size - record_header_max_size;
+    alignment = PCI_DOE_ALIGNMENT;
+    aligned_message_size = (spdm_test_context->test_buffer_size + (alignment - 1)) &
+                           ~(alignment - 1);
+    transport_message_size = aligned_message_size;
```
Comment (jyao1) [inline unit_test/fuzzing/test_transport/test_spdm_transport_pci_doe_encode_message/spdm_transport_pci_doe_encode_message.c:49]: please explain why

**Comment by jyao1** [comment]
Please also explain 
1) Where is the problem? What is the root cause?
2) Why you fix in this way?
3) Why you need change seed?

File: unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c (hunk 3 above)
Comment (yaohuixguo) [reply unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c:60]: uint8_t secured_message[LIBSPDM_MAX_MESSAGE_BUFFER_SIZE];
secured_message requires a section of memory, which cannot be less than the total_secured_message_size
https://github.com/DMTF/libspdm/blob/345e2f866ca463f989343a126e7beb6a563d7c31/library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c#L252-L255

File: unit_test/fuzzing/test_transport/test_spdm_transport_mctp_encode_message/transport_mctp_encode_message.c (hunk 4 above)
Comment (yaohuixguo) [reply unit_test/fuzzing/test_transport/test_spdm_transport_mctp_encode_message/transport_mctp_encode_message.c:49]: There are alignment operations in the function
https://github.com/DMTF/libspdm/blob/345e2f866ca463f989343a126e7beb6a563d7c31/library/spdm_transport_pcidoe_lib/libspdm_doe_pcidoe.c#L71-L76

File: unit_test/fuzzing/test_transport/test_spdm_transport_pci_doe_encode_message/spdm_transport_pci_doe_encode_message.c (hunk 5 above)
Comment (yaohuixguo) [reply unit_test/fuzzing/test_transport/test_spdm_transport_pci_doe_encode_message/spdm_transport_pci_doe_encode_message.c:49]: There are alignment operations in the function
https://github.com/DMTF/libspdm/blob/345e2f866ca463f989343a126e7beb6a563d7c31/library/spdm_transport_mctp_lib/libspdm_mctp_mctp.c#L71-L76

**Comment by yaohuixguo** [comment]
> Please also explain
> 
> 1. Where is the problem? What is the root cause?
> 2. Why you fix in this way?
> 3. Why you need change seed?

1. The problem is on line 255，secured_message_size < total_secured_message_size;
secured_message requires a section of memory, which cannot be less than the total_secured_message_size
https://github.com/DMTF/libspdm/blob/345e2f866ca463f989343a126e7beb6a563d7c31/library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c#L252-L255
2.      - secured_message_size = spdm_test_context->test_buffer_size - record_header_max_size;
        + secured_message_size = sizeof(secured_message);
"total_secured_message_size" will change with the input,  so set "secured_message_size" to the defined maximum size avoid assert.
 app_message_size is spdm_test_context->test_buffer_size - record_header_max_size;
https://github.com/DMTF/libspdm/blob/345e2f866ca463f989343a126e7beb6a563d7c31/library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c#L252-L255

3.(1)The old seed is just "app_message", and the new seed is "app_message + record_header_max".
(2)"spdm_test_context->test_buffer_size - record_header_max_size" as "app_message_size" parameter of function libspdm_encode_secured_message
https://github.com/DMTF/libspdm/blob/345e2f866ca463f989343a126e7beb6a563d7c31/unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c#L62-L64

**Comment by yaohuixguo** [comment]
I have explained，whether I have anything to correct? Thanks

**Pull request closed without merging**